File: 2025/day_2/day2b.js
// https://adventofcode.com/2025/day/2
const fs = require('fs');
let input = fs.readFileSync('./input2.txt', 'utf8');

// input = "11-22,95-115,998-1012,1188511880-1188511890,222220-222224,1698522-1698528,446443-446449,38593856-38593862,565653-565659,824824821-824824827,2121212118-2121212124";

let sum = 0;
const ranges = input.split(",");

const is_invalid_id = (id) => {
	let max_seq_len = Math.floor(id.length / 2);
	for (let len = max_seq_len; len > 0; len--) {

		if (id.length % len === 0) {
			let first = id.slice(0, len), match = false;
			for (let offset = len; offset < id.length; offset += len) {
				let slice = id.slice(offset, offset + len);
				match = slice === first;
				if (!match) break;
			}
			if (match) return true;
		}
	}
};

ranges.forEach(v => {
	const range = v.split("-").map(v => parseInt(v));
	for (let i = range[0]; i <= range[1]; i++) {
		if (is_invalid_id(i.toString())) {
			sum += i;
		}
	}
});

console.log(sum);

File: 2025/day_2/day2b.py
# https://adventofcode.com/2025/day/2
use_example = False
with open("input2.txt" if not use_example else "example2.txt", "r") as file:
	lines = file.read().splitlines()

ranges = [
	tuple(int(n) for n in line.split("-"))
	for line in lines.pop().split(",")
]

def is_invalid_id(pid: str) -> bool:
	str_len = len(pid)
	return any(
		pid == pid[:le] * int(str_len/le)
		for le in range(str_len//2, 0, -1)
		if str_len % le == 0
	)

total = sum([
	pid 
	for start, end in ranges
	for pid in range(start, end+1)
	if is_invalid_id(str(pid))
])

print(total)
# 4174379265